src/internals/tuples/Tuples.ts

```diff
@@ -683,4 +685,46 @@ export namespace Tuples {
     arr1 extends unknown[] | _ | unset = unset,
     arr2 extends unknown[] | _ | unset = unset
   > = Functions.PartialApply<ZipWithFn<fn>, [arr1, arr2]>;
+
+  /**
+   * Range takes a `start` and an `end` integer and produces
+   * a tuple containing integer ranging from `start` to `end`
+   * @param start - the start of the range (included)
+   * @param end - the end of the range (included)
+   * @returns a tuple of integers
+   * @example
+   * ```ts
+   * type T0 = Call<Tuples.Range<3>, 7>; // [3, 4, 5, 6, 7]
+   * type T1 = Eval<Tuples.Range<_, 10>, 5>; // [5, 6, 7, 8, 9, 10]
+   * type T3 = Eval<Tuples.Range< -2, 2>, 5>; // [-2, 1, 0, 1, 2]
+   * type T4 = Eval<Tuples.Range< -5, -2>, 5>; // [-5, -4, -3, -2]
+   * ```
+   */
+  export type Range<
+    start extends number | _ | unset = unset,
+    end extends number | _ | unset = unset
+  > = Functions.PartialApply<RangeFn, [start, end]>;
```

Comment:
❌ I get a "Type instantiation is excessively deep and possibly infinite" here. Anyone knows why?